data.c

```diff
@@ -525,11 +515,66 @@ backup_data_file(const char *from_root,
 
 		file->read_size += read_len;
 	}
+	/*
+	 * In incremental backup mode, append a special 0-filled page, with
```

Comment:
I might be missing something but why do we need a 0-filled BLCKSZ page here? ISTM it's enough to put only the header data with endpoint == ture.

Reply:
I thought that's because pg_rman backup and restore code works with data in PG page units.  But it looks like pg_rman's backup header is written and read independently of the PG page data, so you might be right that just writing the special header without 0-filled PG page is enough.

Reply:
Yeah, at first, I only added header data with endpoint == true, but when I ran pg_rman validate, it detected backup file corruption because of the file size difference.

```
WARNING: size of backup file "base/13307/1249" must be 8192 but 12
WARNING: backup "2018-09-18 10:05:06" is corrupted
```

Reply:
The error message is emitted when the actual backup size is different from the size written in the file list which is genereated by `dir_print_file_list`. So I think if we write the correct size of backup file in the file list we don't get the message.

Reply:
Thanks for your advice!

I've modified my PR only to add a header.

I also faced regression failures using compression option because of the z_stream ends.
(zlib function inflate() return different return code when input ends or continues.)
I added some codes to avoid errors which came from this behavior.